src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,430 @@
+
+### Test Results
+
+#### Client Throughput Comparison
+
+The chart below shows the average client throughput comparison across the schedulers and their respective configurations. The average baseline throughput with both the schedulers was determined to be just over 23K IOPS (4KiB randrw) using fio on the client RBD pool.
+
+
+[![](images/Img1_Throughput_Across_mClock_Profiles_vs_WPQ.png)](images/Img1_Throughput_Across_mClock_Profiles_vs_WPQ.png)
+
+
+<center> Fig. 1: WPQ vs mClock Profiles - Average Client Throughput Comparison With Snaptrim Ops </center>
+
+The client throughput shown is the average throughput reported by fio during the course of the test (300 secs). It is important to note that the snaptrim duration forms a subset of the client I/O run time.
+
+The average client throughput using the WPQ scheduler with default Ceph configuration was 17520.63 IOPS, which is nearly 25% lower than the baseline(WPQ) throughput. But with the mClock scheduler and with the default *high_client_ops* profile, the average client throughput was nearly 10% higher at 19217.59 IOPS when compared to the WPQ scheduler. This meets the minimum reservation criteria of 50% set for client ops in the mClock profile.
+
+The chart also shows the overall client throughput reported with the other mClock profiles. Due to lower reservations allocated to client ops in the other mClock profiles, lower throughputs on the clients are expected. With the *balanced* profile, an average throughput of 18998.55 IOPS (8.4% higher compared to WPQ) was obtained. With the *high_recovery_ops* profile, it was 18293.20 IOPS (4.4% higher compared to WPQ).
+
+This demonstrates that mClock is able to provide the desired QoS for the client with concurrent background snaptrim operations in progress. The average client throughputs with the mClock profiles are higher when compared to the WPQ scheduler.
+
+#### Client Latency Comparison
+
+Fig. 2 below shows the average completion latency (clat) along with the average 95th, 99th and 99.5th percentiles across the WPQ scheduler and the mClock scheduler profiles. In addition, a more accurate measure of the average clat observed during the time snaptrim operations were active is also shown. This is represented as *clat (snaptrim)* in the chart. This shows the effectiveness of mClock profiles over the WPQ scheduler in providing relatively similar average client latencies, but different snaptrim rates. The noticeable differences are in the percentile latencies [95th, 99th, 99.5th] with the *high_client_ops* showing the lowest as expected.
+
+[![](images/Img2_WPQ_vs_mClock_Profile_Client_Pct_Lat.png)](images/Img2_WPQ_vs_mClock_Profile_Client_Pct_Lat.png)
+
+<center> Fig. 2: WPQ vs mClock Profiles - Average Client Latency and Percentile Comparison With Snaptrim Ops </center>
```

Comment:
Same comment here; the caption is right on top of the next paragraph. There are a few other places where the same thing happens.

Reply:
Fixed by adding line breaks. Hopefully it should look better now.